lib/use-tabs.js

```diff
@@ -0,0 +1,77 @@
+import { notifyProperty } from '@neovici/cosmoz-utils/lib/hooks/use-notify-property';
+import {
+	useState, useEffect, useMemo, useCallback
+} from 'haunted';
+import {
+	useHashParam, link
+} from './use-hash-param';
+import {
+	choose, getName, isValid
+} from './utils';
+
+const useTabs = host => {
+	const {
+			selected, hashParam
+		} = host,
+		[tabs, setTabs] = useState([]),
+		param = useHashParam(hashParam),
+		selection = hashParam == null || param == null && selected != null ? selected : param,
+		selectedTab = useMemo(() => choose(tabs, selection), [tabs, selection]);
+
+	useEffect(() => {
+		if (!selectedTab) {
+			return;
+		}
+		selectedTab.toggleAttribute('is-selected', true);
+		const opts = {
+			bubbles: false,
+			composed: true
+		};
+		if (!selectedTab._active) {
+			selectedTab.dispatchEvent(new CustomEvent('tab-first-select', opts));
+			selectedTab._active = true;
+		}
+
+		selectedTab.dispatchEvent(new CustomEvent('tab-select', opts));
+		requestAnimationFrame(() => window.dispatchEvent(new Event('resize')));
+		return () => {
+			selectedTab.toggleAttribute('is-selected', false);
+		};
+
+	}, [selectedTab]);
+
+	useEffect(() => {
+		const onTabAlter = e => {
+			e.stopPropagation();
+			setTabs(prev => prev.slice());
```

Comment:
Based on initial state, and semantics, shouldn't tabs be an array?

Reply:
It is a an array. The state updater can also receive a function to compute a value from previous one.